### PR TITLE
Truncating CommandName to 2048 characters

### PR DESCRIPTION
**Pull request by madeleinefletcher**
Issue #160

### CODE REVIEW

**Comment by SergeyKanzhelev** [comment]
:shipit: thanks! It will help with our backend processing.
@lukekim - you should enforce this limit.

**Comment by AlexBulankou** [comment]
We should do trim() and truncate() to be consistent with Web SDK and also because it is a good thing to do, currently this is doing just truncate()

**Comment by madeleinefletcher** [comment]
Added trimming to sanitization

File: JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts [hunk 1]
```diff
@@ -89,6 +89,7 @@ module Microsoft.ApplicationInsights.Telemetry.Common {
 
         public static sanitizeUrl(url) {
             if (url) {
+                url = Util.trim(url);
                 if (url.length > DataSanitizer.MAX_URL_LENGTH) {
                     url = url.substring(0, DataSanitizer.MAX_URL_LENGTH);
```
Comment (MaxShehovtsov) [inline JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts]: this line of code also looks to be responsible for trimming

Comment (madeleinefletcher) [inline JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts]: Yes, I added that after Alex's comment :)

File: JavaScript/JavaScriptSDK.Tests/CheckinTests/Telemetry/RemoteDependency.tests.ts [hunk 2]
```diff
@@ -38,6 +38,22 @@ class RemoteDependencyTests extends ContractTestHelper {
         });
 
         this.testCase({
+            name: name + "Command name is truncated if too long",
+            test: () => {
+                var urlLength = 2049;
+                var longUrl = "";
+                for (var i = 0; i < urlLength; i++) {
+                    longUrl += "A";
```
Comment (dnduffy) [inline JavaScript/JavaScriptSDK.Tests/CheckinTests/Telemetry/RemoteDependency.tests.ts]: Just for fun I wanted to share these alternatives:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/repeat
http://stackoverflow.com/questions/1877475/repeat-character-n-times

**Comment by AlexBulankou** [comment]
:shipit:

**Comment by dnduffy** [comment]
:+1:

**Pull request merged**